DQM/L1TMonitor/src/L1TCSCTF.cc

```diff
@@ -544,46 +787,159 @@ void L1TCSCTF::analyze(const Event& e, const EventSetup& c)
-              } catch(cms::Exception &) { 
-                bzero(&gblPhi,sizeof(gblPhi)); 
+                gblPhi = srLUTs_[fpga]->globalPhiME(lclPhi.phi_local, lct->getKeyWG(), cscId+1, gangedME11a_);
+              } catch(cms::Exception &) {
+                bzero(&gblPhi,sizeof(gblPhi));
               }
-				
+
               gbletadat gblEta;
               try {
-                gblEta = srLUTs_[fpga]->globalEtaME(lclPhi.phi_bend_local, lclPhi.phi_local, lct->getKeyWG(), cscId+1);
-              } catch(cms::Exception &) { 
-                bzero(&gblEta,sizeof(gblEta)); 
+                gblEta = srLUTs_[fpga]->globalEtaME(lclPhi.phi_bend_local, lclPhi.phi_local, lct->getKeyWG(), cscId+1, gangedME11a_);
+              } catch(cms::Exception &) {
+                bzero(&gblEta,sizeof(gblEta));
+              }
+
+
+              //TrackStub
+              csctf::TrackStub theStub((*lct), (*csc).first);
+              theStub.setPhiPacked(gblPhi.global_phi);
+              theStub.setEtaPacked(gblEta.global_eta);
+
+              float etaG = theStub.etaValue();
+              float phiG = fmod( theStub.phiValue()+15.0*M_PI/180+(sector)*60.0*M_PI/180, 2.*M_PI );
+
+
+              // BX plots
+              // minus side
+              if (endcap == 1 && station == 0 && ring == 1) { csctflcts -> Fill(bx, 8.5); }
+              if (endcap == 1 && station == 0 && ring == 2) { csctflcts -> Fill(bx, 7.5); }
+              if (endcap == 1 && station == 0 && ring == 3) { csctflcts -> Fill(bx, 6.5); }
+              if (endcap == 1 && station == 1 && ring == 1) { csctflcts -> Fill(bx, 5.5); }
+              if (endcap == 1 && station == 1 && ring == 2) { csctflcts -> Fill(bx, 4.5); }
+              if (endcap == 1 && station == 2 && ring == 1) { csctflcts -> Fill(bx, 3.5); }
+              if (endcap == 1 && station == 2 && ring == 2) { csctflcts -> Fill(bx, 2.5); }
+              if (endcap == 1 && station == 3 && ring == 1) { csctflcts -> Fill(bx, 1.5); }
+              if (endcap == 1 && station == 3 && ring == 2) { csctflcts -> Fill(bx, 0.5); }
+
+              // plus side
+              if (endcap == 0 && station == 0 && ring == 1) { csctflcts -> Fill(bx, 9.5); }
+              if (endcap == 0 && station == 0 && ring == 2) { csctflcts -> Fill(bx, 10.5); }
+              if (endcap == 0 && station == 0 && ring == 3) { csctflcts -> Fill(bx, 11.5); }
+              if (endcap == 0 && station == 1 && ring == 1) { csctflcts -> Fill(bx, 12.5); }
+              if (endcap == 0 && station == 1 && ring == 2) { csctflcts -> Fill(bx, 13.5); }
+              if (endcap == 0 && station == 2 && ring == 1) { csctflcts -> Fill(bx, 14.5); }
+              if (endcap == 0 && station == 2 && ring == 2) { csctflcts -> Fill(bx, 15.5); }
+              if (endcap == 0 && station == 3 && ring == 1) { csctflcts -> Fill(bx, 16.5); }
+              if (endcap == 0 && station == 3 && ring == 2) { csctflcts -> Fill(bx, 17.5); }
+
```

Comment:
Okay, I will do what you suggest, I will send it to you before commit, by the way, can I directly edit on this PR, or I need to do another PR?